### PR TITLE
[DC] LDAP connector

### PR DESCRIPTION
**Pull request by greg-snowflake**
Tested by running the UI locally and creating the connection, verifying that the files are ingested properly, and then dropping everything but the stage and creating the connection again using the existing stage flow.

### CODE REVIEW

File: src/connectors/ldap.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        'prompt': "The S3 bucket where the LDAP Logs are collected",
+        'prefix': "s3://",
+        'placeholder': "my-ldap-log-s3-bucket",
+    },
+    {
+        'type': 'str',
+        'name': 'prefix',
+        'title': "Prefix Filter",
+        'prompt': "The folder in LDAP Log Bucket where logs are collected",
+        'default': "ldap/",
```
Comment (andrey-snowflake) [inline src/connectors/ldap.py]: how common is default? is it something only we have?

File: src/connectors/ldap.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        )
+    }
+
+
+def finalize(connection_name):
+    base_name = f'ldap_{connection_name}'
+    pipe = f'data.{base_name}_pipe'
+    table = next(db.fetch(f"SHOW TABLES LIKE '{base_name}_connection' IN data"))
+    options = yaml.load(table['comment'])
+    stage = options.get('existing_stage', f'data.{base_name}_stage')
```
Comment (andrey-snowflake) [inline src/connectors/ldap.py]: I think in order for this to work, you have to add the  `existing_stage` to the `yaml_dump` call above?

File: src/connectors/__init__.py [hunk 3]
```diff
@@ -12,6 +12,7 @@
 from . import tenable_settings
 from . import crowdstrike_devices
 from . import cisco_umbrella
+from . import ldap
```
Comment (andrey-snowflake) [inline src/connectors/__init__.py]: I remember regretting the `okta` without `_log` and we have it elsewhere, so might it make more sense to call this `ldap_logs`? no super strong intuition here, but t might be separate from `ldap_settings` or `ldap_accounts`?

File: src/connectors/ldap.py (hunk 1 above)
Comment (greg-snowflake) [reply src/connectors/ldap.py]: IIRC ldap logs are ingested to S3 via FluentD, so there's not really a meaningful default to put here; it's what we set up when we configured fluentd.

File: src/connectors/ldap.py (hunk 2 above)
Comment (greg-snowflake) [reply src/connectors/ldap.py]: Ah, good catch; the stage I was using adhered to the 'data.{base_name}_stage' format, which is why this made it through testing.

File: src/connectors/__init__.py (hunk 3 above)
Comment (greg-snowflake) [reply src/connectors/__init__.py]: It was azure.py that we regretted (okta.py still exists), but your point is well taken.

**Pull request merged**